### PR TITLE
feat: onConnect w/ controller

### PR DESCRIPTION
**Pull request by ronag**
Part of a larger overhauld of hooks. This on is not breaking and can therefore land earlier.

## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: test/node-test/client-dispatch.js [hunk 1]
```diff
@@ -938,8 +939,9 @@ test('dispatches in expected order for http2', async (t) => {
       onBodySent () {
         dispatches.push('onBodySent')
       },
-      onResponseStarted () {
-        dispatches.push('onResponseStarted')
```
Comment (mcollina) [inline test/node-test/client-dispatch.js:942]: Can you add a test that keeps showing this behavior? Or add a new test for onStart?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: lib/core/request.js [hunk 2]
```diff
@@ -251,8 +251,23 @@ class Request {
     }
   }
 
-  onResponseStarted () {
-    return this[kHandler].onResponseStarted?.()
+  onStart (resume) {
+    try {
+      let pause = false
+
+      // Compatibility with the old API.
```
Comment (metcoder95) [inline lib/core/request.js:258]: Shall we add it to the major checklist?

**Review by metcoder95 — APPROVED** [review]
LGTM, shall we document it and add types for it?

File: lib/core/request.js (hunk 2 above)
Comment (ronag) [reply lib/core/request.js:258]: Already on the list to remove.

**Comment by ronag** [comment]
Updated according to feedback from @BridgeAR

**Pull request closed without merging**